Source/NSAutoresizingMaskLayoutConstraint.m

```diff
@@ -0,0 +1,292 @@
+   version 2.1 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+#import "NSAutoresizingMaskLayoutConstraint.h"
+
+@implementation NSAutoresizingMaskLayoutConstraint
+
+NSUInteger minXAttribute = 32;
+NSUInteger minYAttribute = 33;
+NSUInteger maxXAttribute = 36;
+NSUInteger maxYAttribute = 37;
+
++ (NSArray *)constraintsWithAutoresizingMask:
+                 (NSAutoresizingMaskOptions) autoresizingMask
+                                     subitem: (NSView *) subItem
+                                       frame: (NSRect) frame
+                                   superitem: (NSView *) superItem
+                                      bounds: (NSRect) bounds
+{
+  if (autoresizingMask == NSViewNotSizable)
+    {
+      return [NSArray array];
+    }
+
+  NSAutoresizingMaskLayoutConstraint *xConstraint =
+      [self _xConstraintForAutoresizingMask: autoresizingMask
+                                    subitem: subItem
+                                      frame: frame
+                                  superitem: superItem
+                                     bounds: bounds];
+  NSAutoresizingMaskLayoutConstraint *yConstraint =
+      [self _yConstraintForAutoresizingMask: autoresizingMask
+                                    subitem: subItem
+                                      frame: frame
+                                  superitem: superItem
+                                     bounds: bounds];
+  NSAutoresizingMaskLayoutConstraint *widthConstraint =
+      [self _widthConstraintForAutoresizingMask: autoresizingMask
+                                        subitem: subItem
+                                          frame: frame
+                                      superitem: superItem
+                                         bounds: bounds];
+  NSAutoresizingMaskLayoutConstraint *heightConstraint =
+      [self _heightConstraintForAutoresizingMask: autoresizingMask
+                                         subitem: subItem
+                                           frame: frame
+                                       superitem: superItem
+                                          bounds: bounds];
+
+  NSMutableArray *constraints = [NSMutableArray arrayWithCapacity:4];
+  if ((autoresizingMask & NSViewMinXMargin)
+      && (autoresizingMask & NSViewMaxXMargin))
+    {
+      [constraints addObject: widthConstraint];
+      [constraints addObject: xConstraint];
+    }
+  else
+    {
+      [constraints addObject: xConstraint];
```

Comment:
From just this code, it is unclear why the order changes between the true and false paths of the if statement. Can you add mildly-detailed comments describing the reasoning?

Reply:
Yeah I agree this is strange behaviour. I've added a comment for clarity. It's the observed behaviour from MacOS. I don't think it matters that much for users but some users may have coupled themselves to the ordering, this includes me with my test suite I created for this class.